tests/test_topology.py

```diff
@@ -0,0 +1,165 @@
+import numpy
+import pytest
+
+import lammpsio
+
+
+def test_bonds(snap_top):
+    snap_top.bonds = lammpsio.topology.Bonds(N=6, num_types=2)
+    id = [1, 2, 3, 4, 5, 6]
+    typeid = [1, 1, 1, 2, 2, 2]
+    members = [
+        [1, 2],
+        [2, 3],
+        [3, 4],
+        [5, 6],
+        [6, 7],
+        [7, 8],
+    ]
+
+    snap_top.bonds.id = id
+    snap_top.bonds.typeid = typeid
+    snap_top.bonds.members = members
+
+    assert numpy.allclose(snap_top.bonds.id, id)
+    assert numpy.allclose(snap_top.bonds.typeid, typeid)
+    assert numpy.allclose(snap_top.bonds.members, members)
+
+    # test cases that raise errors
+    with pytest.raises(TypeError):
+        snap_top.bonds = lammpsio.topology.Angles(N=6, num_types=2)
+
+    id = [1, 2, 3, 4, 5, 6, 7]
+    with pytest.raises(TypeError):
+        snap_top.bonds.id = id
+
+    typeid = [1, 1, 1, 2, 2, 2, 3]
+    with pytest.raises(TypeError):
+        snap_top.bonds.typeid = typeid
+
+    members = [[1, 2], [2, 3], [3, 4], [5, 6], [6, 7], [7, 8], [8, 9]]
+    with pytest.raises(TypeError):
+        snap_top.bonds.members = members
+
+
+def test_angles(snap_top):
```

Comment:
There is a lot of duplication between these test cases, which makes sense given the class structure. Is there a way we could consolidate them using a fixture or parametrized test?

For example, we could add a function/fixture that makes the topology for a given N, number of members, and types